langchain/src/chat_models/googlevertexai/common.ts

```diff
@@ -214,19 +228,46 @@ export class BaseChatGoogleVertexAI<AuthOptions>
     return [];
   }
 
-  // TODO: Add streaming support
+  async *_streamResponseChunks(
+    _messages: BaseMessage[],
+    _options: this["ParsedCallOptions"],
+    _runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    // Make the call as a streaming request
+    const instance: GoogleVertexAIChatInstance = this.createInstance(_messages);
+    const parameters = this.formatParameters();
+    const result = await this.streamedConnection.request(
+      [instance],
+      parameters,
+      _options
+    );
+
+    // Get the streaming parser of the response
+    const stream = result.data as GoogleVertexAIStream;
+
+    // Loop until the end of the stream
+    // During the loop, yield each time we get a chunk from the streaming parser
+    // that is either available or added to the queue
+    while (!stream.streamDone) {
+      const output = await stream.nextChunk();
+      const chunk =
+        output !== null
+          ? BaseChatGoogleVertexAI.convertPredictionChunk(output)
+          : new ChatGenerationChunk({
+              text: "",
+              message: new AIMessageChunk(""),
+              generationInfo: { finishReason: "stop" },
```

Comment:
We should break in this case?

Reply:
I did it this way to make sure we have a clear final chunk, but am not opposed to just breaking in this case and not sending it.

Reply:
Oh, no I mean break after getting `null`. It's fine with me to have the final chunk like this but we wouldn't want to receive any more.

Reply:
We need to yield with that chunk, and after that it should be the last thing in the queue or that gets resolved, so the while loop will terminate.